src/mcts/search.cc

```diff
@@ -540,42 +541,43 @@ void Search::EnsureBestMoveKnown() REQUIRES(nodes_mutex_)
 }
 
 // Returns @count children with most visits.
-std::vector<EdgeAndNode> Search::GetBestChildrenNoTemperature(Node* parent,
-                                                              int count) const {
+std::vector<Search::EdgeScore> Search::GetBestChildrenNoTemperature(
+    Node* parent, int count) const {
   MoveList root_limit;
   if (parent == root_node_) {
     PopulateRootMoveLimit(&root_limit);
   }
   // Best child is selected using the following criteria:
+  // * Best terminal score (favor wins, avoid losses).
   // * Largest number of playouts.
   // * If two nodes have equal number:
   //   * If that number is 0, the one with larger prior wins.
   //   * If that number is larger than 0, the one with larger eval wins.
-  using El = std::tuple<uint64_t, float, float, EdgeAndNode>;
+  using El = std::tuple<EdgeScore, uint64_t, float, float>;
   std::vector<El> edges;
   for (auto edge : parent->Edges()) {
     if (parent == root_node_ && !root_limit.empty() &&
         std::find(root_limit.begin(), root_limit.end(), edge.GetMove()) ==
             root_limit.end()) {
       continue;
     }
-    edges.emplace_back(edge.GetN(), edge.GetQ(0), edge.GetP(), edge);
+    edges.emplace_back(EdgeScore(edge), edge.GetN(), edge.GetQ(0), edge.GetP());
```

Comment:
I suppose arguably, with the `EdgeScore::operator<` there doesn't need to be a tuple anymore…

Reply:
Although unless we cache the results of GetN/Q/P in EdgeScore, pulling them out once in the tuple might be slightly faster ??